AsyncNats/NatsConnection.cs

```diff
@@ -482,63 +544,90 @@ private async IAsyncEnumerable<T> InternalSubscribe<T>(string subject, string? q
                 try
                 {
                     await SendUnsubscribe(subscription);
-                    _subscriptions = _subscriptions.Where(s =>s != subscription).ToArray();
+                    _subscriptions.TryRemove(subscription.SubscriptionId, out _);
```

Comment:
Because the ConcurrentDictionary is used, I think the _subscriptionsLock is no longer required.

Reply:
Agreed